drivers/serial/uart_ns16550.c

```diff
@@ -229,6 +233,8 @@ struct uart_ns16550_device_config {
 #if defined(CONFIG_UART_NS16550_ACCESS_IOPORT) || defined(CONFIG_UART_NS16550_SIMULT_ACCESS)
 	bool io_map;
 #endif
+
+	struct reset_dt_spec reset_spec;
```

Comment:
Please use `DT_ANY_INST_HAS_PROP_STATUS_OKAY` to enable reset related stuff. If there is no reset controller attached (which is the majority of cases), there is no need to include all the added code and data.

Reply:
Updated the driver with DT_ANY_INST_HAS_PROP_STATUS_OKAY. Thank you.